packages/lodestar/test/spec/allForks/fork.ts

```diff
@@ -0,0 +1,56 @@
+import {join} from "node:path";
+import {TreeBacked} from "@chainsafe/ssz";
+import {config} from "@chainsafe/lodestar-config/default";
+import {allForks, CachedBeaconStateAllForks} from "@chainsafe/lodestar-beacon-state-transition";
+import {describeDirectorySpecTest} from "@chainsafe/lodestar-spec-test-util";
+import {phase0, altair, bellatrix} from "@chainsafe/lodestar-beacon-state-transition";
+import {ssz} from "@chainsafe/lodestar-types";
+import {ACTIVE_PRESET, ForkName} from "@chainsafe/lodestar-params";
+import {SPEC_TEST_LOCATION} from "../specTestVersioning";
+import {IBaseSpecTest} from "../type";
+import {expectEqualBeaconState, inputTypeSszTreeBacked} from "../util";
+
+export function fork<
+  PreBeaconState extends phase0.BeaconState | altair.BeaconState,
+  PostBeaconState extends altair.BeaconState | bellatrix.BeaconState
+>(pre: ForkName, fork: ForkName): void {
```

Comment:
Here you can use
```
Exclude<ForkName, ForkName.phase0>
```
to prevent having to declare an empty function in `upgradeStateByFork`

https://www.typescriptlang.org/docs/handbook/utility-types.html#excludeuniontype-excludedmembers